### ISSUE TITLE
haxelib server is case sensitive, but haxelib changes case

### ISSUE THREAD

**Issue by haxiomic** [HaxeFoundation/haxelib#465]
See https://github.com/haxiomic/console.hx/issues/7

The haxelib client fails to upgrade a haxelib package with a capital letter – the haxelib client tries to download the package with a lowercase but the server will only serve the file if the casing is correct

The client requests the file `console,hx-0,2,12.zip` but it fails because file on the server is `Console,hx-0,2,12.zip`

Would it be possible to disable case-sensitivity on the haxelib file server?

I don't seem to be able to change the casing of the library now it has been submitted, so it's stuck as `Console.hx`

**Comment by haxiomic** [comment]
Related: https://github.com/HaxeFoundation/haxelib/issues/392#issuecomment-518290028

**Comment by RealyUniqueName** [comment]
I can't figure out why is it capitalized on the server.
haxelib.json always had lowercased name. Did you maybe upload a zip archive with the file name "Console.hx.zip"?

**Comment by haxiomic** [comment]
@RealyUniqueName yeah, when I first uploaded it’s likely I used Console.hx.zip but since then I’ve only used lowercase

**Comment by haxiomic** [comment]
Hey @RealyUniqueName, what's the situation with this bug now after your change? Am I safe to push an update to Console.hx or is there a separate server change that needs to be done first?

**Comment by RealyUniqueName** [comment]
I don't remember any details, but looking at the change server does not need any updates.

**Comment by haxiomic** [comment]
Thanks for all your investigations @tobil4sk! This is very comprehensive and helpful

To me ideally libs are just handle case insensitive all the time

**Comment by tobil4sk** [comment]
Another problem has emerged: what capitalisation should the library flag have? (the `-D console.hx` in the case of Console.hx). Currently it depends on the capitalisation of the library name with the `-lib` flag. We either need to check the `haxelib.json` (which potentially doesn't exist, which is allowed currently) to use the official library capitalisation or we need to specify that it should be in lowercase.

**Comment by tobil4sk** [comment]
The local client issues are not relevant to the server, so I moved those details to a separate issue: #529

**Comment by haxiomic** [comment]
Very good point, thanks for flagging up
